server/gendb/table.go.tmpl

```diff
@@ -883,6 +886,87 @@ func (t {{ $modelName }}Table) scan{{ $modelName }}sBy{{ $computedIndexName }}(c
   return err
 }
 {{- end }}
+
+{{- if transactionsEnabled $xdbConfig }}
+{{- range $transactWithSchemaName := $xdbConfig.EnableTransactions }}
+{{- $transactWithModelName := pascalize $transactWithSchemaName }}
+func (t {{ $modelName }}Table) transactSave{{ $modelName }}And{{ $transactWithModelName }}(ctx context.Context, m1 models.{{ $modelName }}, m1ConditionMap map[string]interface{}, m2 models.{{ $transactWithModelName }}, m2ConditionMap map[string]interface{}) error {
+  data1, err := encode{{ $modelName }}(m1)
+  if err != nil {
+    return err
+  }
+
+  m1CondExpr, m1ExprVals, m1ExprNames, err := buildCondExpr(m1ConditionMap)
+  if err != nil {
+    return err
+  }
+
+  data2, err := encode{{ $transactWithModelName }}(m2)
+  if err != nil {
+    return err
+  }
+
+  m2CondExpr, m2ExprVals, m2ExprNames, err := buildCondExpr(m2ConditionMap)
+  if err != nil {
+    return err
+  }
+
+	input := &dynamodb.TransactWriteItemsInput{
+		TransactItems: []*dynamodb.TransactWriteItem{
+			{
+				Put: &dynamodb.Put{
+          TableName: aws.String(t.name()),
+					Item:      data1,
+					ConditionExpression:       m1CondExpr,
+					ExpressionAttributeValues: m1ExprVals,
+					ExpressionAttributeNames:  m1ExprNames,
+				},
+			},
+			{
+				Put: &dynamodb.Put{
+    			TableName: aws.String(fmt.Sprintf("%%s-{{ $transactWithModelName }}s", t.Prefix)),
+					Item:      data2,
+					ConditionExpression:       m2CondExpr,
+					ExpressionAttributeValues: m2ExprVals,
+					ExpressionAttributeNames:  m2ExprNames,
+				},
+			},
+		},
+	}  
+  _, err = t.DynamoDBAPI.TransactWriteItemsWithContext(ctx, input)
+
+  return err
+}
+{{- end }}
+
+func buildCondExpr(conditionMap map[string]interface{}) (*string, map[string]*dynamodb.AttributeValue, map[string]*string, error) {
+	var condExpr *string
+	var exprVals map[string]*dynamodb.AttributeValue
+	var exprNames map[string]*string
+	if conditionMap != nil && len(conditionMap) > 0 {
```

Comment:
Nit: I think you can omit the nil check, if you can `len()` on a slice it just returns 0.